### ISSUE TITLE
`NullPointerException` on creating a repository on S3 with IAM role authentication

### ISSUE THREAD

**Issue by hammerhead** [crate/crate#14595]
### CrateDB version

5.4.2

### CrateDB setup information

Single-node EC2 instance with an IAM role attached:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "s3:ListAllMyBuckets",
                "s3:GetBucketLocation"
            ],
            "Resource": "*"
        },
        {
            "Sid": "VisualEditor1",
            "Effect": "Allow",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::crate-tutorials",
                "arn:aws:s3:::crate-tutorials/*"
            ]
        }
    ]
}
```

### Problem description

Creating a snapshot repository isn't possible when using authentication through an attached IAM role.

### Steps to Reproduce

Verify the role works on the EC2 instance:
```bash
[ec2-user@ip-172-31-29-98 ~]$ aws s3 ls s3://crate-tutorials
                           PRE indices/
                           PRE weather-data/
2022-12-13 09:37:38  556900250 power_consumption_dataset.json
2023-08-03 11:22:46  877067119 train-00000-of-00004-1a1932c9ca1c7152.json.gz
```

Attempting to create the repository on the same EC2 instance:
```sql
cr> CREATE REPOSITORY test TYPE s3 WITH ("bucket" = 'crate-tutorials');
RepositoryVerificationException[[test] Unable to verify the repository, [test] is not accessible on master node: NullPointerException 'Cannot invoke "com.amazonaws.auth.AWSCredentials.getAWSAccessKeyId()" because "this.credentials" is null']

org.elasticsearch.repositories.RepositoryVerificationException: [test] Unable to verify the repository, [test] is not accessible on master node: NullPointerException 'Cannot invoke "com.amazonaws.auth.AWSCredentials.getAWSAccessKeyId()" because "this.credentials" is null'
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1187)
	at org.elasticsearch.repositories.RepositoriesService$3.doRun(RepositoriesService.java:252)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

### Actual Result

See above.

### Expected Result

Successful creation of the repository as stated in the [documentation](https://crate.io/docs/crate/reference/en/5.4/sql/statements/create-repository.html#s3):
> If you are using Amazon S3 in conjunction with [IAM roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html), the access_key and secret_key parameters must be left undefined.

**Comment by jeeminso** [comment]
Thanks for reporting @hammerhead , NPE should be handled by #14686 but fixing this will uncover the original behaviour which is another exception:
```
RepositoryVerificationException[[test] Unable to verify the repository, [test] is not accessible on master node: AmazonS3Exception 'Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: K1XM9DBRJ090R; S3 Extended Request ID: hGMrBUr+ColeA76HPNhNi7ys4Lfk1T/iRwSdJdWmzwwKrjyYpQYI0ogVjhHomU=; Proxy: null)']
[2023-09-13T01:05:08,631][DEBUG][o.e.c.s.MasterService    ] [Pierre Avoi] took [0s] to notify listeners on successful publication of cluster state (version: 19, uuid: H13Vl5k8yzS_yAmCk3fw) for [delete_repository [test]]
```
could you double check if you see the same? This should be reproducible with `5.3.1`.

**Comment by hammerhead** [comment]
Hi @jeeminso,

I tested on 5.3.1 and didn't see any exception there, the snapshot gets created successfully (restore also works):

```sql
cr> SELECT VERSION();
+----------------------------------------------------------------------------------------------------------------------------------+
| pg_catalog.version()                                                                                                             |
+----------------------------------------------------------------------------------------------------------------------------------+
| CrateDB 5.3.1 (built 8ef4c2b/2023-04-28T16:51:45Z, Linux 6.1.41-63.114.amzn2023.x86_64 amd64, OpenJDK 64-Bit Server VM 19.0.2+7) |
+----------------------------------------------------------------------------------------------------------------------------------+
SELECT 1 row in set (0.003 sec)
cr> CREATE REPOSITORY test TYPE s3 WITH ("bucket" = 'crate-tutorials');
CREATE OK, 1 row affected (0.285 sec)
cr> CREATE SNAPSHOT test.s1 ALL;
CREATE OK, 1 row affected (0.120 sec)
```
```bash
[ec2-user@ip-172-31-29-98 ~]$ aws s3 ls s3://crate-tutorials
                           PRE indices/
2023-09-13 06:12:45        496 index-0
2023-09-13 06:12:46          8 index.latest
2023-09-13 06:12:45        227 meta-uOwAcxlsQXKq5_L4in7GIw.dat
2023-09-13 06:12:45        222 snap-uOwAcxlsQXKq5_L4in7GIw.dat
```

**Comment by jeeminso** [comment]
Thank you @hammerhead

**Comment by jeeminso** [comment]
Confirmed fixed with `crate-5.5.0-202309170026-5ba7838`:
```
cr> CREATE REPOSITORY test TYPE s3 WITH ("bucket" = '...');
[2023-09-17T00:53:18,686][INFO ][o.e.r.RepositoriesService] [Schinhorn] put repository [test]
CREATE OK, 1 row affected (2.035 sec)
```
